File: src/2515.shortest-distance-to-target-string-in-a-circular-array.2598/solution.ts
function closetTarget (words: string[], target: string, startIndex: number): number {
    if (words[startIndex] === target) {
        return 0;
    }
    const n = words.length;
    let left = (startIndex - 1 + n) % n;
    let right = (startIndex + 1) % n;
    let dis = 1;
    while (dis <= n / 2) {
        if (words[left] === target || words[right] === target) {
            return dis;
        }
        dis++;
        left = (left - 1 + n) % n;
        right = (right + 1) % n;
    }
    return -1;
}

File: src/2515.shortest-distance-to-target-string-in-a-circular-array.2598/solution.java
class Solution {
    public int closetTarget(String[] words, String target, int startIndex) {
        int result = words.length;
        for(int i=0;i< words.length;i++){
            if(!words[i].equals(target)){
                continue;
            }
            int diff = Math.abs(startIndex-i);
            result = Math.min(result, Math.min(diff, words.length-diff) );
        }
        return result == words.length?-1:result;
    }
}